### PR TITLE
Allow annotation selector to match annotation values

### PR DESCRIPTION
**Pull request by liggitt**
Fixes server-side matching for #10877

CLI specification of annotation selector still needs fixing

### CODE REVIEW

**Comment by liggitt** [comment]
cc @deads2k

**Comment by smarterclayton** [comment]
LGTM but want david's sign off.  Will move to 1.3.0 before i cut the release.

**Comment by smarterclayton** [comment]
[test]

File: pkg/quota/api/helpers.go [hunk 1]
```diff
@@ -19,13 +19,9 @@ func GetMatcher(selector ClusterResourceQuotaSelector) (func(obj runtime.Object)
...
 
-	var annotationSelector labels.Selector
+	var annotationSelector map[string]string
 	if len(selector.AnnotationSelector) > 0 {
-		var err error
-		annotationSelector, err = unversioned.LabelSelectorAsSelector(&unversioned.LabelSelector{MatchLabels: selector.AnnotationSelector})
-		if err != nil {
-			return nil, err
-		}
+		annotationSelector = selector.AnnotationSelector
```
Comment (liggitt) [inline pkg/quota/api/helpers.go]: Do we need to make a copy of the map here?

Comment (smarterclayton) [inline pkg/quota/api/helpers.go]: Probably to be safe.

**Comment by deads2k** [comment]
lgtm

**Comment by liggitt** [comment]
[merge]

**Pull request merged**